### PR TITLE
Addon to update the ports in alt-svc header in reverse mode

### PR DESCRIPTION
**Pull request by errorxyz**
#### Description

Alt-Svc headers in HTTP responses advertise the host to use while migrating from HTTP1/2 to HTTP3 (amongst other information). This addon modifies the hosts specified in this header to our reverse proxy target.
refs #7025 

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

**Comment by errorxyz** [comment]
Notes for self:
Connecting to a remote reverse proxy works[^1] via chrome but not using firefox
Connecting to a locally hosted reverse proxy does not work in chrome and firefox
[^1]: works as in browsers attempt to connect to the updated port in the alt-svc header

File: mitmproxy/addons/update_alt_svc.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+from mitmproxy import ctx
+from mitmproxy.http import HTTPFlow
+from mitmproxy.proxy import mode_specs
+
+ALT_SVC = "alt-svc"
+PORT_PATTERN = r'(?<=:)\d{1,5}(?=")'
+
+
+def update_alt_svc_port(data: str, port: int) -> str:
+    return re.sub(PORT_PATTERN, f"{port}", data)
```
Comment (mhils) [inline mitmproxy/addons/update_alt_svc.py:12]: I think we need to override the optional host override as well, no?

File: mitmproxy/addons/update_alt_svc.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+
+class UpdateAltSvc:
+    def load(self, loader):
+        loader.add_option(
+            "update_alt_svc",
```
Comment (mhils) [inline mitmproxy/addons/update_alt_svc.py:18]: `keep_alt_svc_header` to match `keep_host_header`?

**Review by mhils — COMMENTED** [review]
Looks good so far. 👍

**Comment by errorxyz** [comment]
Do we want an option for the user to specify the host value that will be used to modify the `Alt-Svc` header?

**Comment by errorxyz** [comment]
> Notes for self: Connecting to a remote reverse proxy works[1](#user-content-fn-1-9cee89b6c97433b5ee61df40af6e38df) via chrome but not using firefox[2](#user-content-fn-2-9cee89b6c97433b5ee61df40af6e38df) Connecting to a locally hosted reverse proxy does not work in chrome and firefox

I can swear that firefox randomly started working for a locally hosted reverse proxy after hours of unsuccessful debugging.

**Comment by mhils** [comment]
> Do we want an option for the user to specify the host value that will be used to modify the `Alt-Svc` header?

Do we have a good use case for that? For esoteric use cases they can always turn the option on and write their own addon (or use modify_headers).

**Review by mhils — APPROVED** [review]

**Pull request merged**